src/main/java/no/nav/fo/veilarbdialog/rest/DialogRessurs.java

```diff
@@ -23,89 +23,98 @@
 import static java.lang.Math.toIntExact;
 import static java.util.Collections.singletonList;
 import static java.util.stream.Collectors.toList;
-import static no.nav.apiapp.util.StringUtils.notNullOrEmpty;
-import static no.nav.apiapp.util.StringUtils.of;
-import static no.nav.fo.veilarbdialog.service.AutorisasjonService.erEksternBruker;
-import static no.nav.fo.veilarbdialog.service.AutorisasjonService.erInternBruker;
-import static no.nav.fo.veilarbdialog.util.FunksjonelleMetrikker.nyDialogBruker;
-import static no.nav.fo.veilarbdialog.util.FunksjonelleMetrikker.nyDialogVeileder;
-
-@Component
-@Import({
-        RestMapper.class,
-        KontorsperreFilter.class
-})
-public class DialogRessurs implements DialogController, VeilederDialogController {
-
-    @Inject
-    private AppService appService;
-
-    @Inject
-    private RestMapper restMapper;
-
-    @Inject
-    private Provider<HttpServletRequest> requestProvider;
+import static no.nav.fo.veilarbdialog.auth.AuthService.erEksternBruker;
+import static no.nav.fo.veilarbdialog.auth.AuthService.erInternBruker;
+
+@RestController
+@RequestMapping(
+        value = "/api/dialog",
+        consumes = MediaType.APPLICATION_JSON_VALUE,
+        produces = MediaType.APPLICATION_JSON_VALUE
+)
+@RequiredArgsConstructor
+@Slf4j
+public class DialogRessurs {
+
+    private final DialogDataService dialogDataService;
+    private final RestMapper restMapper;
+    private final HttpServletRequest httpServletRequest;
+    private final KontorsperreFilter kontorsperreFilter;
+    private final AuthService auth;
+    private final KladdService kladdService;
+    private final FunksjonelleMetrikker funksjonelleMetrikker;
+
+    @GetMapping
+    public List<DialogDTO> hentDialoger() {
 
-    @Inject
-    private KontorsperreFilter kontorsperreFilter;
+        try {
+            return dialogDataService.hentDialogerForBruker(getContextUserIdent())
+                    .stream()
+                    .filter(dialog -> kontorsperreFilter.harTilgang(auth.getSsoToken(), dialog.getKontorsperreEnhetId()))
```

Comment:
Kaster exception for eksternbrukere? Eller misforstår jeg nå? Uansett, risikabelt å mekke på abac ting iom at det er vanskelig å teste.

Reply:
Vel, APIet til `Pep` har endra seg og krever nå en ident.

Hvis du ser på gamle `PepClient.java:186` så ser du noe sånt som

`return this.buildRequest().withSamlToken((String)SecurityUtils.getSamlToken().orElse((Object)null)).withOidcToken((String)SecurityUtils.getOidcToken().orElse((Object)null));`

og det er altså det vi prøver å få til her. Dvs. hente ut tilsvarende som `withSamlToken(...)`.

Det er altså ikke mulig å ikke endre denne koden ved overgang til nye `commons-java`.